receiver/prometheusreceiver/internal/instance_relabel_end_to_end_test.go

```diff
@@ -0,0 +1,224 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal_test
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/processor/batchprocessor"
+	"go.opentelemetry.io/collector/service"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver"
+)
+
+func TestJobInstanceRelabelEndToEnd(t *testing.T) {
```

Comment:
This should probably happen in the `prometheusreceiver` package where we already have a framework for e2e tests and shape testing of the resulting pdata metrics.

Reply:
Hrm, so, I tried moving it there, but the framework there is about verifying the "scrape" output and I cannot figure out how to verify the target and job labels that are added. Could you give me a pointer?

Reply:
You'd want to build a [test like this one](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b00283b724e83bb91526108cdc3c43520c167961/receiver/prometheusreceiver/metrics_receiver_labels_test.go#L31), probably in that same file since it's already testing labeling handling.  The [validation function](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b00283b724e83bb91526108cdc3c43520c167961/receiver/prometheusreceiver/metrics_receiver_labels_test.go#L47) can prepare a [set of expected resource attributes](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b00283b724e83bb91526108cdc3c43520c167961/receiver/prometheusreceiver/metrics_receiver_labels_test.go#L47), which in this case would include the job and instance ID values mapped to the appropriate resource attribute identifiers.  `doCompare()` will [take care of asserting](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/519ea7ff61cb09d2be2d40b070fad753a1603a9b/receiver/prometheusreceiver/metrics_receiver_helper_test.go#L326-L332) that all of the expected resource attributes are present.

Reply:
Ah ack, working on it now. I didn't realise the resource attributes part.